test/api/dockerflow.test.ts

```diff
@@ -72,17 +72,26 @@ describe('dockerflow endpoints', () => {
     });
 
     it('answers an error to the heartbeat if the bucket does not exist', async () => {
-      jest
-        .spyOn(Bucket.prototype, 'exists')
+      // We can't use jest.spyOn here because of a typescript issue with jest.
+      // When we use jest.spyOn, the typescript compiler complains that the
+      // mock return value must be void. This is happening because apparently
+      // typescript doesn't understand the overloaded Bucket.prototype.exists method.
```

Comment:
Isn't the problem that `exists` isn't in the types for the gogole storage library? I can see it in the js files but not the type files. Could we improve the types locally?

Reply:
No, google storage library is a typescript project so it includes types already. `exists` is in here: 
https://github.com/googleapis/nodejs-storage/blob/21f173eb17d4216e2b42ffdd1ed0104aeda7cf13/src/nodejs-common/service-object.ts#L323-L353
This is from the `ServiceObject` base class. [`Bucket` extends from this one: ](https://github.com/googleapis/nodejs-storage/blob/21f173eb17d4216e2b42ffdd1ed0104aeda7cf13/src/bucket.ts#L684)
The issue is that Typescript doesn't work well with the overloaded exists functions above. There is a github issue also in the commit message if you are interested.